### ISSUE TITLE
Describe the process to make a smooth transition for OA

### ISSUE THREAD

**Issue by ERosendo** [freelawproject/courtlistener#3013]
I'm opening this issue to discuss the steps to deploy the new OA search feature and avoid downtime:

- [ ] Create a new env variable called `ELASTICSEARCH_DSL_AUTO_REFRESH` and set it to `False`

**Comment by mlissner** [comment]
We'll also need a command to ingest the existing content?

**Comment by albertisfu** [comment]
For indexing Audio documents we could use the built-in command. Only for indexing the OA Alerts well need the custom command described in https://github.com/freelawproject/courtlistener/issues/3014

**Comment by albertisfu** [comment]
Here is the revised process to deploy OA Search PR to production and achieve a smooth transition to ES:

**- Before merging OA PR and deploying to prod.**
- Set the required environment variable for Courtlistener.
- Waffle environment variables to ensure flags are disabled by default when performing the deployment. This ensures that Solr Search will continue in use:
`WAFFLE_FLAG_DEFAULT = False`
`WAFFLE_SWITCH_DEFAULT = False`

- Elasticsearch env vars:
Number of shards and replicas for OA Search index:
```
ELASTICSEARCH_OA_NUMBER_OF_SHARDS
ELASTICSEARCH_OA_NUMBER_OF_REPLICAS
```

Number of shards and replicas for OA Alerts Percolator index:
```
ELASTICSEARCH_OA_ALERTS_NUMBER_OF_SHARDS
ELASTICSEARCH_OA_ALERTS_NUMBER_OF_REPLICAS
```
Disable elasticsearch auto-refresh in prod:
`ELASTICSEARCH_DSL_AUTO_REFRESH = False`

- Load synonyms and stopwords files into Elasticsearch cluster:

Create a ConfigMap for storing the synonyms and stopwords files:

**elastic-dictionaries.yaml**
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: elastic-dictionaries
data:
  synonyms_en.txt: |
    # Paste synonyms_en.txt content here.
  stopwords_en.txt: |
    # Paste stopwords_en.txt content here.
```
Apply the ConfigMap:
`kubectl apply -f elastic-dictionaries.yaml`

Update elastic-cluster manifest:

```
- name: elasticsearch
        volumeMounts:
        - name: elastic-dictionaries-volume
          mountPath: /usr/share/elasticsearch/config/dictionaries
volumes:
        - name: elastic-dictionaries-volume
          configMap:
            name: elastic-dictionaries
```
Update the cluster:
`kubectl apply -f elastic-cluster.yml`

**- Merge and deploy OA PR.**

**- After deploy:**

- Create indexes:
`search_index --create --models audio.Audio`
`search_index --create --models alerts.Alert`

 - Populate indices:
Oral Arguments:
`search_index --populate --models audio.Audio --parallel`
 
 - OA Alerts Percolator index:
`cl_index_search_alerts --pk-offset 0 --alert-type oa`

At this point, existing Audio documents and Alerts for OA will be indexed, while new objects will be indexed by the `es_signal_processor`.

- After indexing is finished:

Verify `oa-es-active` flag is in `Unknown` status for everyone and active for Superusers. This will allow us to inspect and manual test OA Search to ensure everything is functioning correctly.

Flip flag and switch:
- Enable `oa-es-active` flag  and `oa-es-alerts-active` switch for everyone, so OA search and alerts are handled by Elasticsearch.

**- Keep Solr running?**
  - At this point, OA search and OA Alerts will be working with ES, while Audio documents will continue to be indexed in Solr as well.
  - It may be a good idea to keep Solr OA indexing running for a few days just in case of a contingency related to Elasticsearch. This way, we can fall back on Solr while solving any problems related to ES.
  - Once OA Search and Alerts are in a stable state, we can disable Solr OA documents indexing and remove flags and switch code relative to OA.

**Comment by albertisfu** [comment]
An additional task we need to do is setting up cron jobs for sending ES Search Alerts (for now, only OA) at daily, weekly, and monthly intervals, once all the OA Alerts are indexed.

```
cl_send_scheduled_alerts --rate dly
cl_send_scheduled_alerts --rate wly
cl_send_scheduled_alerts --rate mly
```

**Comment by mlissner** [comment]
This all went very smoothly. Three little things:

1. The percolator index was already created, probably by somebody saving alert (me?), so we had to delete it and re-create it, so it would have the right fields.

2. The elastic dictionary wasn't quite right, as fixed in #3088. 

3. Some trouble getting indentation right for the k8s config, and we forgot to switch k8s contexts before deploying the configmap. 

Otherwise, very smooth. Closing.
